### PR TITLE
Parse and Write Marteria in PDF

### PR DESCRIPTION
**Pull request by t-bullock**
Does it make sense for the mode and annotations to simply have x,y coords in the xml instead?

### CODE REVIEW

**Comment by jdaly101** [comment]
Yeah, that's fine. The XML file format was just me trying out some things and it's totally up for grabs. I want the front end to generate XML files eventually, so hopefully users will never have to mess with them.

File: pdfmaker/kassia.py [hunk 1]
```diff
@@ -44,10 +45,15 @@ def setDefaults(self):
...
 
-        psalticaTTF = "fonts/EZ Psaltica.TTF"
-        oxeiaTTF    = "fonts/EZ Oxeia.ttf"
-        omegaTTF    = "fonts/EZ Omega.ttf"
+        psalticaTTF     = "fonts/EZ Psaltica.TTF"
+        specialOneTTF   = "fonts/EZ Special-I.TTF"
+        specialTwoTTF   = "fonts/EZ Special-II.TTF"
+        oxeiaTTF        = "fonts/EZ Oxeia.ttf"
+        omegaTTF        = "fonts/EZ Omega.ttf"
+
```
Comment (jdaly101) [inline pdfmaker/kassia.py:53]: This is good for now. Eventually let's move this off into a separate file that registers font and throws up error if any are missing.

File: pdfmaker/kassia.py [hunk 2]
```diff
@@ -44,10 +45,15 @@ def setDefaults(self):
         self.pageAttrib['line_height'] = 72
         self.pageAttrib['line_width'] = self.pageAttrib['paper_size'][0] - (self.pageAttrib['left_margin'] + self.pageAttrib['right_margin'])
```
Comment (jdaly101) [inline pdfmaker/kassia.py:46]: So I'm thinking that the default attributes will probably end up being paper size dependent. That's a PR for another day, but let me know what you think.

File: pdfmaker/sample.xml [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
-            84
-        </rightannotation>
-        <rightannotation>
+        <mode color="#cf232b" font_size="20" font="EZSpecial-II" align="center" top_margin="30">
+            h q T
+        </mode>
+        <annotation color="#cf232b" font_size="20" font="EZSpecial-I" align="right" translate="true" top_margin="-30">
+            s
+        </annotation>
+        <annotation color="#1111ff" font_size="12" font="EZOmega" align="right">
```
Comment (jdaly101) [inline pdfmaker/sample.xml:13]: Is there anything different between how you handle a mode and how you handle an annotation?

Comment (t-bullock) [inline pdfmaker/sample.xml:13]: Hmm, I guess the only real difference is that the default font will probably be different (Special-II for the EZ fonts). But that's not a big deal. Assuming Kassia supports other fonts at some point, we can't say for sure that the mode would always need a different font. It probably makes most sense to just treat it as an annotation.

**Comment by t-bullock** [comment]
Any ideas on how best to specify two separate fonts for the same element? The marteria piece is not quite complete until it can use two different fonts (at least for EZ), and eventually we'll need to specify which neumes use Special-I and II. Maybe each neume needs to be its own element?

File: pdfmaker/kassia.py (hunk 2 above)
Comment (t-bullock) [inline pdfmaker/kassia.py:46]: Hmm, I'll have to think about that. It seems like page margin and space between lines of music would be the same no matter the size of the page, but maybe there are some attributes that I'm not thinking of. Also, ReportLab allows for any sized page, right? Would we just create defaults for common paper sizes?

Comment (jdaly101) [inline pdfmaker/kassia.py:46]: If someone wanted to create Zoe style books, the default margins and spacing would probably be a little smaller. I'd say it's pretty low priority, tho. Yes, Reportlab has a bunch of preset sizes (A4, legal, letter) and lets you customize it, but we'd only need defaults for a couple of them with a catch-all (probably 1" margins).

**Comment by jdaly101** [comment]
Yeah, the mode is going to be awkward. Maybe we can just specify a font family (e.g. EZ-family) for now and automatically generate the neumes. It shouldn't be multiple elements, imo, because there's no compelling reason to put the complexity on the user.

**Comment by jdaly101** [comment]
You're good to merge this, btw. The mode should probably be a separate PR.

**Comment by t-bullock** [comment]
Maybe we can get around this by using BZ Byzantia instead of EZ Psaltica? Looks like the same fonts, only they're broken up into font files that might make more sense (Marteria pieces all contained in one file).

**Pull request merged**